### ISSUE TITLE
ACE3 items stop mine detector from working

### ISSUE THREAD

**Issue by nsgundy** [acemod/ACE3#879]
Since ACE3 items like `ACE_fieldDressing` are currently defined as mine detectors with a detection range of `-1`, they stop mine detectors that have been added to the inventory later on from working (they do not provide the sought after increase in detection range).

See [BI ticket 23921](http://feedback.arma3.com/view.php?id=23921#c92044) on how to reproduce.
[BI Forum](http://forums.bistudio.com/showthread.php?191290) post that this started with.

ACE3 Version: 3.0.0.1 at commit 28a8fbc

**Mods:**
- @cba_a3
- @ace3

**ACE3 modules loaded:**
- ace_common
- ace_dragging
- ace_interact_menu
- ace_interaction
- ace_main
- ace_medical
- ace_modules

**Placed ACE3 Modules:**
- None

**Description:**
the function used to figure out a unit's mine detection range, is currently stopping at the first "Mine Detector" item it can find and uses its detection range, rather than parsing through the whole inventory to find the item with the best detection range.

**Steps to reproduce:**
In editor with VA map loaded, place a APERS Tripwire Mine (named "mine") 30m directly north of the player, player facing directly south, and once spawned in, running backwards (only holding down S). Run the following script to display the distance to the mine until detected:

``` sqf
[] spawn {
  while {!(mine mineDetectedBy side player)} do {
    hintSilent formatText ["%1",player distance mine];
  };
};
```

If, after spawning in I execute

``` sqf
removeAllItems player;
player addItem "MineDetector";
player addItem "ACE_fieldDressing";
```

and then walk backwards (just holding down `S`) towards the mine while watching the distance to the mine decrease (it will stop decreasing once it has been detected), I do get a detection range of 15m for both the Marksman and the Explosive Specialist.

If however I change the order and have the ACE item added first

``` sqf
removeAllItems player;
player addItem "ACE_fieldDressing";
player addItem "MineDetector";
```

both the Marksman and the Explosive Specialist will not detect the mine before stepping onto it.

**Where did the issue occur?**
Editor

**Comment by alganthe** [comment]
please use (this form) [http://ace3mod.com/wiki/user/how-to-report-an-issue.html]

**Comment by commy2** [comment]
Nice find.
Well better than no VA or all items being mine detectors I guess.

I hope we can replace the vanilla mine detector with something scripted in future.

**Comment by commy2** [comment]
Fucking mine detectors.

**Comment by alganthe** [comment]
Definitely an issue but you can't really do much about it.

**Comment by nsgundy** [comment]
Well, I see two courses of action:
* Get BI to give us a way to define an item as just an item and still show up in Arsenal
* Get BI to change their function to figure out the detection range (i.e. do not stop at the first item that is defined as a mine detector or exclude mine detectors with a negative detection range)
* Replace the mine detector with something better that doesn't use the BI mine detection framework

**Comment by alganthe** [comment]
* Get BI to give us a way to define an item as just an item and still show up in Arsenal
 I guess we can make a cross on that one

* Get BI to change their function to figure out the detection range (i.e. do not stop at the first item that is defined as a mine detector or exclude mine detectors with a negative detection range)

Would be the best solution ATM

* Replace the mine detector with something better that doesn't use the BI mine detection framework

A potential short term fix, to be replaced by the second option.

**Comment by nsgundy** [comment]
Here is a link to a discussion on the AGM GitHub, pretty much discussing various ways of implementing a mine-detector replacement: https://github.com/KoffeinFlummi/AGM/issues/735

**Comment by nsgundy** [comment]
See [BI ticket 23925](http://feedback.arma3.com/view.php?id=23925) for request to have a "Misc. Item" classification that works with Arsenal. Hope this is in everyone's interest.

**Comment by nsgundy** [comment]
Good news on BI ticket 23925, Mr. Moricky left [this note](http://feedback.arma3.com/view.php?id=23925#c92202):
> We're currently planning an engine solution for misc item interaction. Until then, I'm afraid there is no temporary fix planned.

**Comment by commy2** [comment]
:+1:

**Comment by olanguedoc** [comment]
any update on this? My EOD just exploded.

**Comment by commy2** [comment]
Can't fix this. I'd link the Feauture request from the BI tracker, but ... yeah.

**Comment by synixebrett** [comment]
This can be closed right? Since they now use CBA_MiscItem which is a bipod item.

**Comment by jonpas** [comment]
Fixed by #5502

@synixebrett thanks for noticing!

**Comment by commy2** [comment]
>Fucking mine detectors.

woke
